### PR TITLE
new post

### PR DESCRIPTION
**Pull request by lbenie**
First technical post

### CODE REVIEW

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+They are available for most languages like CSS, Python, JavaScript, HTML, etc...
+
+## Eslint
+
+[Eslint](http://eslint.org/) is the most recent of the four linting tools
+previously mentioned. It was created by **Nicholas C. Zakas** in june 2013.
+
+> Its goal is to provide a pluggable linting utility for JavaScript.
+
+Design to be heavily extensible, it comes with a large set of custom rules and
```
Comment (malaporte) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Designed

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+When writing JavaScript, I spend a lot of time fixing simple mistakes. Unlike
+compiled languages you are more likely to make mistakes. It is easy for syntax
+errors to sneak into your code without realizing it until you actually try and
+run your code.
+
+How many times have I got an `undefined` variable because I refactored some code
+and forgot to rename *that* variable.
+
+Even tough it has been more than 5 years since I wrote my first `Hello World`.
```
Comment (marcantoineveilleux) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: though

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Since `ESLint` is extremely extensible, I have created a [shareable config rule
+set](https://github.com/coveo/eslint-config-coveo) to help my fellow colleagues
+here at [Coveo](www.coveo.com) who write Pure JavaScript.
+
+### Open-sourcing
+
+`eslint-config-coveo` started as an internal project for my team (Salesforce
+integration) then we decided -- Hey why not open-source it?
+
+With that in mind, I created a gulp task wrapper that use **our** rule set
```
Comment (marcantoineveilleux) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: that uses

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+tags:
+  - JavaScript
+  - Lint
+  - Eslint
+  - Code style
+
+author:
+  name: Lucien Bénié
+  bio: Salesforce integration wizard
```
Comment (marcantoineveilleux) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Lol :) gj

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+and forgot to rename *that* variable.
+
+Even tough it has been more than 5 years since I wrote my first `Hello World`.
+The feeling remains the same -- Why did I make this mistake *again* ?
+
+<!-- more -->
+
+To help me fix some of those mistakes, I tried a few linting tools over the
+years. From the overly strict `JSLint` to the more flexible variant `JSHint` and
+`JSCS`. I recently discovered `Eslint` and fell in love with its
```
Comment (wfortin) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Might want to link to all of these tools.

Comment (wfortin) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: You should write `ESLint`

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+gulp.task('lint', function() {
+  gulp
+    .src('src/**/*.js')
+    .pipe(linter());
+});
+```
+
+Sit back, relax and enjoy watching your silly mistakes from your terminal
+output!
```
Comment (wfortin) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Might want to give a little heads about the typescript counter part? pretty-typescript ?

**Comment by wfortin** [comment]
Great article @lbenie !

**Comment by pastjean** [comment]
@lbenie update la date pour aujourd'hui et release nous ça !

**Comment by lbenie** [comment]
@pastjean @wfortin @malaporte @marcantoineveilleux 

Done ;)

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+previously mentioned. It was created by **Nicholas C. Zakas** in june 2013.
+
+> Its goal is to provide a pluggable linting utility for JavaScript.
+
+Designed to be heavily extensible, it comes with a large set of custom rules and
+it is really easy to install. It gives precise and concise output by including
+the rule name out of the box. You are always aware of which rule was causing an
+error.
+
+`Eslint` provides good documentation for its rules. The rules list is easy to
```
Comment (francoislg) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: ESLint

Comment (francoislg) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Don't quite understand what "The rules list" references. The rules list in the documentation ? 
Maybe change for something more precise ("Its rules list") ? 

And add a link to that doc.

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+`eslint-config-coveo` started as an internal project for my team (Salesforce
+integration) then we decided -- Hey... why not open-source it?
+
+With that in mind, I created a gulp task wrapper that uses **our** rule set
+defined from [that shareable
+config](https://github.com/coveo/eslint-config-coveo). You can find the project
+`pretty-javascript` [here](https://github.com/coveo/pretty-javascript).
+
+**Having said all that**
```
Comment (francoislg) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: Don't really like "Having said all that", sounds a little too french-y. :wink: 

I think a simple "Now what?" or "I want it too!" would be catchier.

File: _posts/2016-06-16-open-source-coveo-javascript-linter.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+[here](https://github.com/coveo/pretty-typescript) (huge kudos to Dominique
+Bégin).
+
+Similarly to *pretty-javascript*, you only have to include `pretty-typescript` and `gulp` from `npm`.
+
+```sh
+npm install --save-dev pretty-typescript gulp
+```
+
+Then simply add a task to lint your TypeScript code in your `gulpfile` as follow :
```
Comment (francoislg) [inline _posts/2016-06-16-open-source-coveo-javascript-linter.md]: "as follows: " or "as the following: "

**Comment by francoislg** [comment]
:gg: Interesting stuff :)

**Pull request merged**